Source/Csla.Blazor.WebAssembly/State/SessionManager.cs

```diff
@@ -40,35 +40,44 @@ public class SessionManager(
     /// </summary>
     public Session GetSession()
     {
-      _session ??= [];
-      _session.LastTouched = DateTimeOffset.UtcNow;
+      if (_session == null)
+      {
+        _session ??= [];
+        _session.LastTouched = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+      }
       return _session;
     }
 
     /// <summary>
     /// Updates the current user's session 
-    /// data locally and on the web server.
+    /// data locally.
     /// </summary>
     /// <param name="session">Current user session data</param>
     public void UpdateSession(Session session)
     {
-      _session.LastTouched = DateTimeOffset.UtcNow;
+      _session.LastTouched = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
       _session = session;
     }
 
     /// <summary>
     /// Retrieves the current user's session from
-    /// the web server to the wasm client.
+    /// the web server to the wasm client
+    /// if SyncContextWithServer is true.
     /// </summary>
     public async Task<Session> RetrieveSession()
     {
       if (_options.SyncContextWithServer)
       {
-        if (_session == null)
+        long lastTouched = 0;
+        if (_session != null)
+          lastTouched = _session.LastTouched;
+        var url = $"{_options.StateControllerName}?lastTouched={lastTouched}";
+
+        var stateResult = await client.GetFromJsonAsync<StateResult>(url);
+        if (stateResult.ResultStatus == ResultStatuses.Success)
         {
-          var result = await client.GetFromJsonAsync<byte[]>(_options.StateControllerName);
-          var formatter = new MobileFormatter(ApplicationContext);
-          var buffer = new MemoryStream(result)
+          var formatter = SerializationFormatterFactory.GetFormatter(ApplicationContext);
```

Comment:
Inject the serializer directly instead of going through the static factory (which get's it from the DI anyway).

Reply:
That's a good idea - but a bigger change because the serializer isn't currently provided as a DI service. Maybe something for a future version.